compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -897,6 +897,24 @@ impl<'a, 'tcx> Visitor<'tcx> for BoundVarContext<'a, 'tcx> {
                         })
                         .unzip();
                 self.record_late_bound_vars(hir_id, binders.clone());
+
+                for param in bound_generic_params {
+                    match param.kind {
+                        GenericParamKind::Lifetime { .. } => {}
+                        GenericParamKind::Type { default, .. } => {
+                            if let Some(ty) = default {
+                                self.visit_ty(ty);
+                            }
+                        }
+                        GenericParamKind::Const { ty, default } => {
+                            self.visit_ty(ty);
+                            if let Some(default) = default {
+                                self.visit_body(self.tcx.hir().body(default.body));
+                            }
+                        }
+                    }
+                }
```

Comment:
This is generally the correct fix.

Though I feel like this could be merged with the same logic for handling lifetimes in const defaults in item generics (#93669). Maybe you could move the `visit_ty` and `visit_body` calls into `visit_generic_param`, and then this can be replaced with:

```suggestion
                walk_list!(self, visit_generic_param, bound_generic_params);
```